### PR TITLE
added support for converting final states in conjunction of negations

### PR DESCRIPTION
**Pull request by martinhruska**
This PR implements conversion of final states defined as conjunction of negations to set of final states. This is a case of explicit NFA created by mintermization from bitvector NFA where final states where defined by such formula.

### CODE REVIEW

File: src/nfa/tests-nfa.cc [hunk 1]
```diff
@@ -907,6 +907,32 @@ TEST_CASE("Mata::Nfa::construct() from IntermediateAut correct calls")
...
+                "q5 a7 q7\n";
+
+        const auto auts = Mata::IntermediateAut::parse_from_mf(parse_mf(file));
+        inter_aut = auts[0];
+
+        construct(&aut, inter_aut, &symbol_map);
+        REQUIRE(aut.final.size() == 4);
+        REQUIRE(is_in_lang(aut, encode_word(symbol_map, {"a1", "a2"})));
+        REQUIRE(is_in_lang(aut, encode_word(symbol_map, {"a1", "a2", "a3"})));
+        REQUIRE(is_in_lang(aut, encode_word(symbol_map, {"a1", "a2", "a3"})));
```
Comment (jurajsic) [inline src/nfa/tests-nfa.cc:933]: Why is this test here twice? Maybe add a test whether word "a1a2a4" is not in the language instead.

**Review by jurajsic — APPROVED** [review]
It looks ok to me, just some small thing in one test.

File: src/nfa/tests-nfa.cc (hunk 1 above)
Comment (martinhruska) [reply src/nfa/tests-nfa.cc]: Done.

**Review by tfiedor — APPROVED** [review]
No issues.

File: include/mata/inter-aut.hh [hunk 2]
```diff
@@ -229,6 +231,19 @@ public:
...
 
+    bool are_final_states_conjunction_of_negation() const
+    {
+        return is_graph_conjunction_of_negations(final_formula);
+    }
+
+    static bool is_graph_conjunction_of_negations(const FormulaGraph& graph);
+
+    /**
+     * Methods returns a set of final state in the case that they were entered as a conjunction
```
Comment (Adda0) [inline include/mata/inter-aut.hh:242]:
```suggestion
     * Method returns a set of final states in the case that they were entered as a conjunction
```
Typo.

**Review by Adda0 — APPROVED** [review]
The PR looks good to me.

**Comment by martinhruska** [comment]
Thanks for the reviews.

**Pull request merged**